src/viam/trajex/totg/trajectory.cpp

```diff
@@ -1266,10 +1357,70 @@ trajectory trajectory::create(class path p, options opt, integration_points poin
                         break;
                     }
 
+                    // Probe the candidate next point to check if it would exceed limit curves.
```

Comment:
I don't like that this whole block is duplicated here now. I suspect that the order of operations in forward following should be rotated around so this happens more naturally.

That, or, maybe even better, is to get rid of the forward following state entirely, and have the forward integrating pass itself be able to decide whether it is on the velocity curve or not.

That'd make implementing the bisection logic easier too, because we would only need it in one place.